x509/src/pkix_oids.rs

```diff
@@ -188,3 +189,278 @@ pub const PKIX_OCSP_NOCHECK: ObjectIdentifier = ObjectIdentifier::new("1.3.6.1.5
+        PKIX_CE_SUBJECT_ALT_NAME => "subjectAltName",
+        PKIX_CE_ISSUER_ALT_NAME => "issuerAltName",
+        PKIX_CE_BASIC_CONSTRAINTS => "basicConstraints",
+        PKIX_CE_CRLNUMBER => "cRLNumber",
+        PKIX_CE_CRLREASONS => "cRLReasons",
+        PKIX_CE_ISSUINGDISTRIBUTIONPOINT => "issuingDistributionPoint",
+        PKIX_CE_DELTACRLINDICATOR => "deltaCRLIndicator",
+        PKIX_CE_CERTIFICATEISSUER => "certificateIssuer",
+        PKIX_CE_HOLDINSTRUCTIONCODE => "holdInstructionCode",
+        PKIX_HI_HOLDINSTRUCTION_CALLISSUER => "holdinstruction-callissuer",
+        PKIX_HI_HOLDINSTRUCTION_REJECT => "holdinstruction-reject",
+        PKIX_CE_INVALIDITYDATE => "invalidityDate",
+        PKIX_CE_NAME_CONSTRAINTS => "nameConstraints",
+        PKIX_CE_CRL_DISTRIBUTION_POINTS => "cRLDistributionPoints",
+        PKIX_CE_CERTIFICATE_POLICIES => "certificatePolicies",
+        PKIX_CE_POLICY_MAPPINGS => "policyMappings",
+        PKIX_CE_AUTHORITY_KEY_IDENTIFIER => "authorityKeyIdentifier",
+        PKIX_CE_POLICY_CONSTRAINTS => "policyConstraints",
+        PKIX_CE_FRESHEST_CRL => "freshestCrl",
+        PKIX_CE_INHIBIT_ANY_POLICY => "inhibitAnyPolicy",
+        PKIX_OCSP_NOCHECK => "ocspNoCheck",
+        PIV_NACI_INDICATOR => "pivNaciIdendicator",
+        PKIXALG_RSA_ENCRYPTION => "rsaEncryption",
+        PKIXALG_EC_PUBLIC_KEY => "ecPublicKey",
+        PKIXALG_DH => "ecDH",
+        PKIXALG_SECP192R1 => "secp192r1",
+        PKIXALG_SECP163K1 => "sect163k1",
+        PKIXALG_SECP163R2 => "sect163r2",
+        PKIXALG_SECP224R1 => "secp224r1",
+        PKIXALG_SECP233K1 => "sect233k1",
+        PKIXALG_SECP233R1 => "sect233r1",
+        PKIXALG_SECP256R1 => "secp256r1",
+        PKIXALG_SECP283K1 => "sect283k1",
+        PKIXALG_SECP283R1 => "sect283r1",
+        PKIXALG_SECP384R1 => "secp384r1",
+        PKIXALG_SECP409K1 => "sect409k1",
+        PKIXALG_SECP409R1 => "sect409r1",
+        PKIXALG_SECP521R1 => "secp521r1",
+        PKIXALG_SECP571K1 => "sect571k1",
+        PKIXALG_SECP571R1 => "sect571r1",
+        PKIXALG_ECDSA_WITH_SHA224 => "ecdsa-with-SHA224",
+        PKIXALG_ECDSA_WITH_SHA256 => "ecdsa-with-SHA256",
+        PKIXALG_ECDSA_WITH_SHA384 => "ecdsa-with-SHA384",
+        PKIXALG_ECDSA_WITH_SHA512 => "ecdsa-with-SHA512",
+        PKIXALG_SHA224_WITH_RSA_ENCRYPTION => "sha224WithRSAEncryption",
+        PKIXALG_SHA256_WITH_RSA_ENCRYPTION => "sha256WithRSAEncryption",
+        PKIXALG_SHA384_WITH_RSA_ENCRYPTION => "sha384WithRSAEncryption",
+        PKIXALG_SHA512_WITH_RSA_ENCRYPTION => "sha512WithRSAEncryption",
+        PKIXALG_RSAES_OAEP => "RSAES-OAEP",
+        PKIXALG_PSPECIFIED => "pSpecified",
+        PKIXALG_MGF1 => "MFG1",
+        PKIXALG_RSASSA_PSS => "sha1",
+        PKIXALG_SHA224 => "sha224",
+        PKIXALG_SHA256 => "sha256",
+        PKIXALG_SHA384 => "sha384",
+        PKIXALG_SHA512 => "sha512",
+        _ => "",
+    }
+}
```

Comment:
I really don't like this function. It smells like OpenSSL. What is the purpose of it?

Reply:
Logging and display.